### DISCUSSION TITLE
Gradle: Building several artifacts from a single module

### DISCUSSION THREAD

**Discussion by mschorsch** [quarkusio/quarkus#37471] [Q&A]
In Maven i can build several artifacts from a single module (https://quarkus.io/guides/maven-tooling#maven-multi-build).

How to do it with Gradle? In the Quarkus-Gradle documentation i could not find anything.

[a comment by an automated account is not shown]

**Comment by brianwyka** [comment]
@mschorsch, did you ever figure this out?

**Reply by mschorsch** [reply]
Unfortunately I have not found a solution.

cc @aloubyansky any Idea?

**Reply by aloubyansky** [reply]
What would be an advantage over creating a subproject/module per application instead? Especially if there differences in dependencies.

**Reply by brianwyka** [reply]
Leveraging the `@IfBuildProfile` annotation can allow a single module setup while sharing code easily and without the need for minimum of 3 modules with 2 of them depending on a "core" module.

**Reply by mschorsch** [reply]
> What would be an advantage over creating a subproject/module per application instead? Especially if there differences in dependencies.

The primary use case is that I want to build the same application with different configurations without repeating the entire build. This is sometimes necessary because some parameters are fixed at build time.
